data-prepper-plugins/kafka-source/README.md

```diff
@@ -0,0 +1,150 @@
+# Kafka source
+
+This is the Data Prepper Kafka source plugin that reads records from Kafka topic. It uses the consumer API provided by Kafka to read messages from the broker.
+
+The Kafka source plugin supports OpenSearch 2.0.0 and greater.
+
+## Usages
+
+The Kafka source should be configured as part of Data Prepper pipeline yaml file.
+
+## Configuration Options
+
+```
+log-pipeline:
+  source:
+    kafka:
+      bootstrap_servers:
+        - 127.0.0.1:9092,localhost:9093, 127.0.0.1:9094
+      topics:
+        - topic:
+            name: my-topic-2
+            auth_type: SASL_OAUTH #optional
+            consumer_group:
+              group_name: kafka-consumer-group-2
+              group_id: DPKafkaProj-2
+              workers: 10 #optional and default is 10
+              autocommit: false  #optional and dafault is false
+              autocommit_interval: 5s  #optional and dafault is 5s
+              session_timeout: 45s  #optional and dafault is 45s
+              max_retry_attempts: 1000 #optional and dafault is 5
+              max_retry_delay: 1s #optional and dafault is 5
+              auto_offset_reset: earliest  #optional and dafault is earliest
+              thread_waiting_time: 1s  #optional and dafault is 1s
+              max_record_fetch_time: 4s #optional and dafault is 4s
+              heart_beat_interval: 3s  #optional and dafault is 3s
+              buffer_default_timeout: 5s  #optional and dafault is 5s
+              fetch_max_bytes: 52428800  #optional and dafault is 52428800
+              fetch_max_wait: 500  #optional and dafault is 500
+              fetch_min_bytes: 1  #optional and dafault is 1
+              retry_backoff: 100s  #optional and dafault is 10s
+              max_poll_interval: 300000s  #optional and dafault is 300000s
+              consumer_max_poll_records: 500  #optional and dafault is 500
+            schema:
+              registry_url: http://localhost:8081/
+              key_deserializer: org.apache.kafka.common.serialization.StringDeserializer
+              value_deserializer: org.apache.kafka.common.serialization.StringDeserializer
+              schema_type: plaintext
```

Comment:
schema_type: plaintext
 record_type: plaintext

If these are optional and user chooses to not give them, default logic should be plaintext

Reply:
Resolved.
by default, the schema_type is PLAINTEXT